src/Stack/Package.hs

```diff
@@ -734,14 +734,14 @@ libraryFiles lib = do
         resolveFilesAndDeps
             Nothing
             (dirs ++ [dir])
-            (names <> exposed)
+            names
```

Comment:
My reasoning for not having it be a Set is that there is cost to constructing the Set, and we always convert it from / to a list.  It is semantically more correct, though, indeed!